### PR TITLE
add interface-mtu option

### PR DESCRIPTION
**Pull request by alvin-huang**
The main motivation for this option is to support allowing jumbo frames. However, instead of adding a boolean for jumbo frames and setting 'option interface-mtu 9000' when the boolean is true, I believe giving the user an option to pick a desired MTU is better.

If mtu==undef, the option will not be added to the dhcpd.conf file and dhcp will just use the default 1500 MTU. If desired, we can set this default value and add the line 'option interface-mtu 1500' by default to the template. However, from what I have seen in the past, users normally have the option not set at all if it is just the default.

### CODE REVIEW

File: templates/dhcpd.pool.erb [hunk 1]
```diff
@@ -60,6 +60,9 @@ subnet <%= @network %> netmask <%= @mask %> {
 <% elsif @search_domains -%>
   option domain-search "<%= @search_domains.split(/[, ]+/).join('", "') %>";
 <% end -%>
+<% if @mtu -%>
```
Comment (domcleal) [inline templates/dhcpd.pool.erb]: This parameter will need adding to `dhcp::pool` too.

**Comment by domcleal** [comment]
If you could add a test to spec/classes/init_spec.rb and spec/defines/pool_spec.rb too, that'd be appreciated.

**Comment by ekohl** [comment]
It can also be validated as an integer:
```puppet
if $mtu {
  validate_integer($mtu)
}
```

**Comment by alvin-huang** [comment]
@ekohl @domcleal thanks for the suggestions. How does this look?

**Comment by ekohl** [comment]
You can add the same validation to pool.pp, but otherwise :+1:

**Comment by alvin-huang** [comment]
Misordered the output in my tests originally but now it should all be good :)

**Comment by mmoll** [comment]
@alvin-huang please add the validation also to the pool, as it can be overridden also there, :+1: otherwise

**Comment by alvin-huang** [comment]
@mmoll done!

**Comment by mmoll** [comment]
merged, thanks @alvin-huang!

**Pull request merged**